### ISSUE TITLE
Daemon holds ~1.2 GB resident, peaks above 4 GB; peak is unexplained

### ISSUE THREAD

**Issue by flyingrobots** [flyingrobots/AI-TTS#22]
## Measured, so the numbers are not guesses

The daemon holds ~1.2 GB resident and has been observed peaking above 4 GB.
Measured in a fresh process:

| Stage | Peak RSS |
|---|---|
| interpreter + `aitts` imported | 191 MB |
| after the Kokoro model loads | **1218 MB** |
| after one clip is synthesised | 1422 MB |
| + a second language pipeline | 1483 MB |
| + a third | 1509 MB |
| + a fourth | 1542 MB |

A live daemon idle for 19 hours reported `phys_footprint: 1182 MB` with
`phys_footprint_peak: 4473 MB`.

## What is by design

The ~1 GB step is the torch runtime plus the 82M-parameter model, and holding
it resident is the entire premise of the daemon — architecture §2 chose a
long-lived model-resident process precisely so the queue rather than the engine
governs perceived responsiveness. Reloading per utterance is the failure mode
this exists to avoid.

Per-language pipelines are **not** the problem: about 30–75 MB each, so all six
languages would add well under half a gigabyte. Bounding that cache would be
optimising the wrong thing.

Note also that `ps` RSS is badly misleading here. The idle daemon showed 50 MB
RSS while its `phys_footprint` was 1182 MB — macOS had compressed or evicted
almost the entire model. Use `footprint -p <pid>`, which is what Activity
Monitor reports.

## What is not explained

`phys_footprint_peak: 4473 MB` is roughly three times the steady state and is
not accounted for by the pipeline cache. Candidates, in order of plausibility:

1. **Concurrent synthesis.** `workers=2` by default, so two clips can be in the
   engine at once, each with its own activations. A long document splits into
   many chunks and keeps both workers busy for its whole length.
2. **Transient during load.** Reading the `.pth` allocates the state dict and
   the model at the same time, so the load itself can momentarily need roughly
   double the resident size.
3. Something genuinely unbounded that this investigation did not find.

Worth resolving before assuming (1). The daemon now reports timings through
`metrics`, but nothing reports memory, so this is currently unanswerable from
the outside.

## Possible work, roughly in order of value

- **Report it.** Add resident and peak footprint to the `metrics` op. Cheap,
  and it turns this issue from anecdote into a series.
- **Attribute the peak.** Synthesise a long multi-chunk document with
  `workers=1` and `workers=2` and compare peaks. If concurrency is the cause,
  the worker count becomes a memory/throughput dial worth documenting.
- **Consider unloading after idle.** Drop the pipelines and the model after N
  minutes of silence, reload on next use. Trades first-clip latency after a
  quiet spell for a much smaller idle footprint. Note macOS already compresses
  the idle model away, so the benefit is mostly to memory *pressure* rather
  than to a number in Activity Monitor — and it directly contradicts
  "model resident", so it should be opt-in if it happens at all.

## Not the cause

Ruled out during the investigation: no leak in the bounded structures
(`MetricsRecorder` caps at 256 samples, the CoreAudio UID map is per device,
the resolved-asset map is per file), and gradients are already disabled
upstream via `@torch.no_grad()`.

A real defect *was* found while measuring this — the model was running in
training mode with nine live dropout layers — but that was a quality bug rather
than a memory one, and it is fixed.

**Comment by flyingrobots** [comment]
## Data point: the peak counter is per process, which narrows this

Measured on a freshly restarted daemon (pid changed, so `phys_footprint_peak`
reset), after warmup and two real clips spoken end to end:

```
phys_footprint:      1404 MB
phys_footprint_peak: 1404 MB   <- equal to current
```

Compare the daemon that had been up 19 hours: `phys_footprint: 1182 MB`,
`phys_footprint_peak: 4473 MB`.

So a single clip on a warm model peaks at 1404 MB and the peak equals the
steady state — the model load itself is **not** what reaches 4 GB. The 4473 MB
figure accumulated across 19 hours of real use.

That demotes candidate (2) in the issue body (transient double allocation
during load) and promotes (1), concurrent synthesis of a long multi-chunk
document across two workers.

The experiment to settle it is now well defined: synthesise one long document
with `workers=1` and again with `workers=2`, reading `footprint -p <pid>` after
each. If the peak scales with worker count, the worker count is a
memory/throughput dial and should be documented as one.

Timings from the same run, for context — the queue is not the cost:

| | |
|---|---|
| synthesis wait | 2672 ms |
| synthesis itself | 2667 ms |
| playback wait | **1 ms** |
